upnp_port_forward/client.py

```diff
@@ -96,16 +154,16 @@ def _find_internal_ip_on_device_network(upnp_dev: upnpclient.upnp.Device) -> str
     raise _NoInternalAddressMatchesDevice(parsed_url.hostname)
 
 
-WAN_SERVICE_NAMES = (
-    "WANIPConn1",
-    "WANIPConnection.1",  # Nighthawk C7800
-    "WANPPPConnection.1",  # CenturyLink C1100Z
-    "WANPPPConn1",  # Huawei B528s-23a
-)
+def _get_wan_service(
+    upnp_dev: upnpclient.upnp.Device, wan_service_name: Optional[str]
```

Comment:
I think that we should look at doing this such that it **either** uses a set of explicitly provided names *or* it falls back to the default names by default. 

My intuition is that combining the provided name with the set of default names is going to be problematic because 1) it doesn't give any guarantees that the port mapping will be setup on the provided service name and 2) it doesn't allow us to use the API in a way that excludes the default list which may be desirable.

So we want to allow explicitely specifying one or more names, and if those are not expicitely provided we fall back to the known set of service names.